metabase/metabase
# Saved question selection UX

Status - Need help from @daltojohnso to help sort out blockers 

## What this does
Updates the picker you use to select a saved question as a starting point for further exploration to properly account for nested collections and provides more space overall to work.

![image](https://user-images.githubusercontent.com/5248953/108872229-f38d5680-75c7-11eb-8419-9d20e407a73e.png)


How to test

1. Go to "Ask a question" and select "Saved questions"  (see bug below if you only have one database locally)
2. You should see any collections containing saved questions that can be used to start a new question on the left.
3. Clicking on any one of those should show saved questions that are valid starting points for new questions 
4. It should work with large numbers of collections and saved questions

--- 
Project planning scratch (still organizing things here)

Remaining work that has to be done before merge
- [ ] The picker only works for simple mode right now. Not sure how best to also make sure it'll show up in the notebook.
- [ ] Both the collection list and the saved question list probably need to be wrapped in a virtualized list wrapper.
- [ ] Need to test independent scrolling of both the sidebar and list with many items
- [ ] Not sure if we're handling the "no saved questions" case here yet.
- [ ] Handle situation where a parent collection might not have a valid saved q but a child does.
- [ ] There's a "Loading..." state for the entire popover that is quite gross if you hit it (intermittent). We'll need to restructure the markup a bit to have the overall layout stay relatively similar to the final product to prevent flashing.
- [ ] Collection item expand arrows aren't spaced properly

Related issues:
- https://github.com/metabase/metabase/issues/11806 - second bullet point
